pkg/util/flag/flags.go

```diff
@@ -50,7 +50,6 @@ type IPVar struct {
 // Set sets the flag value
 func (v IPVar) Set(s string) error {
 	if len(s) == 0 {
-		v.Val = nil
```

Comment:
I think this should be `*v.Val = ""`. Same below.

Setting the value to a `nil` pointer is incorrect (if this function is called again it would panic below), so the code is just wrong. Also, it would not actually set the pointer to `nil` because this method has value receiver. Two bugs on one line :)

One more thing I'd fix:
```go
	_ pflag.Value = IPVar{}
	_ pflag.Value = IPPortVar{}
```

Methods are implemented on value not on the pointer, so assertion should be for the value.

`ReservedMemoryVar` suffers from the same issue - sets the pointer to `nil`. Another inconsistency is that it returns an error where other methods here panic. Should probably panic too.

Reply:
Thanks a lot for your feedback and the additional recommendations, @ash2k!! I aimed for the simplest approach that fixes the staticcheck issues and doesn't change the methods' behavior. Agree with you, though. It has probably been the original author's intention to allow setting empty values. Addressed your feedback in 94d3f4c.